agent/config/builder.go

```diff
@@ -1324,6 +1325,10 @@ func (b *Builder) Validate(rt RuntimeConfig) error {
 		return fmt.Errorf("auto_encrypt.allow_tls can only be used on a server.")
 	}
 
+	if rt.ServerMode && rt.AdvertiseReconnectTimeout != 0 {
+		return fmt.Errorf("advertise_reconnect_timeout can only be used on a client")
```

Comment:
Probably need a FlagsAndEdgeCases test to test this.

Reply:
I added the test case to ensure it errors out as expected.